src/bin/stratisd.rs

```diff
@@ -85,20 +123,45 @@ fn run() -> StratisResult<()> {
                          revents: 0,
                          events: libc::POLLIN,
                      });
-            1
+            2
         }
-        None => 0,
+        None => 1,
     };
 
     // Don't timeout if eventable lets us be event-driven
-    let poll_timeout = if engine_fds_end_idx != 0 { -1 } else { 10000 };
+    let poll_timeout = if engine_fds_end_idx != 1 { -1 } else { 10000 };
 
     loop {
+        // Process any udev block events
+        if fds[0].revents != 0 {
+            loop {
+                match udev.receive_event() {
+                    Some(event) => {
+                        if let Some(added) = handle_udev_add(&event) {
```

Comment:
I would think you could destructure the tuple right here, as:

```
if let Some((devnode, device)) = handle_udev_add(&event) {
```

but maybe not.